public/app/features/alerting/unified/components/rules/RuleDetails.tsx

```diff
@@ -53,6 +55,38 @@ export const RuleDetails: FC<Props> = ({ rule }) => {
   );
 };
 
+interface EvaluationBehaviorSummaryProps {
+  rule: CombinedRule;
+}
+
+const EvaluationBehaviorSummary = ({ rule }: EvaluationBehaviorSummaryProps) => {
+  let forDuration: string | undefined;
+  let every: string | undefined;
+
+  if (isGrafanaRulerRule(rule.rulerRule)) {
+    forDuration = rule.rulerRule.for;
+    every = rule.group.interval;
+  } else if (isAlertingRulerRule(rule.rulerRule) && rule.rulerRule.for) {
+    forDuration = rule.rulerRule.for;
+    every = rule.group.interval;
+  }
```

Comment:
What's the difference between these two? I don't see any

Reply:
Ah yes there isn't, this is another one of those typescript things – I'll change the logic here but it needs a check for `rulerRule` to be an object (because for some reason it's typed as optional)

Reply:
Actually, only recording rules don't have a `for` duration so I've changed the logic to reflect that 👌